### PR TITLE
Fixes and new features

### PR DESCRIPTION
**Pull request by botn365**
makes rocket not take damge from non players
add api to make fluids seen as rocket fuel and tier them
made fueler faster
the lander will now drop with the rocket fueled with the rest of thefuel instead of puting the rest of the fuel in the lander
using a launchcontroller will now reduse countdown and instently teleport the rocket

### CODE REVIEW

**Comment by 0lafe** [comment]
Do we need to remove the launching part?

**Comment by botn365** [comment]
the idea behind it is to use it as a way to teleport stuff and launching takes up a lot of time and is prety buggy

**Comment by mitchej123** [comment]
> Do we need to remove the launching part?

Yes.  It's buggy AF, people die, rockets poof, all around unpleasant.

**Comment by Dream-Master** [comment]
@mitchej123 @bartimaeusnek can this be merged?

File: src/main/java/micdoodle8/mods/galacticraft/core/recipe/RecipeManagerGC.java [hunk 1]
```diff
@@ -61,10 +64,21 @@ public static void loadRecipes()
     @SuppressWarnings("unchecked")
     private static void addUniversalRecipes()
     {
-    	Object meteoricIronIngot = ConfigManagerCore.recipesRequireGCAdvancedMetals ? GCItems.meteoricIronIngot : "ingotMeteoricIron";
+
+        if (Loader.isModLoaded("miscutils"))
+        {
+            RocketFuelRecipe.addFuel("fluid.rocketfuelmixa",8);
```
Comment (draknyte1) [inline src/main/java/micdoodle8/mods/galacticraft/core/recipe/RecipeManagerGC.java:70]: I think you’ve got the max tiers for the fuels wrong.

File: src/main/java/micdoodle8/mods/galacticraft/core/recipe/RecipeManagerGC.java [hunk 2]
```diff
@@ -61,10 +64,21 @@ public static void loadRecipes()
     @SuppressWarnings("unchecked")
     private static void addUniversalRecipes()
     {
-    	Object meteoricIronIngot = ConfigManagerCore.recipesRequireGCAdvancedMetals ? GCItems.meteoricIronIngot : "ingotMeteoricIron";
+
+        if (Loader.isModLoaded("miscutils"))
```
Comment (bartimaeusnek) [inline src/main/java/micdoodle8/mods/galacticraft/core/recipe/RecipeManagerGC.java:68]: Hard no aslong as these are the only higher fuel tier options.

**Review by bartimaeusnek — CHANGES_REQUESTED** [review]

File: src/main/java/micdoodle8/mods/galacticraft/core/recipe/RecipeManagerGC.java (hunk 2 above)
Comment (botn365) [reply src/main/java/micdoodle8/mods/galacticraft/core/recipe/RecipeManagerGC.java:68]: got any sugestions to add to the list of fuels

**Comment by GeicowithaBango** [comment]
cetane-boosted and maybe high octane gasoline?

**Comment by GTNH-Colen** [comment]
> cetane-boosted and maybe high octane gasoline?

Super against forcing players to set up the entire HOG line just so they can launch their rocket. Especially given that by the time they will need it they are well beyond oil based energy. Why not use helium plasma or something for the high ones?

**Comment by botn365** [comment]
there can be mutiple high tier fuels

**Comment by GeicowithaBango** [comment]
Oh, it was not supposed to be a gate, just a more dense rocket fuel, so that you dont have to load your rocket every single launch etc.
Certain Plasma as fuel is a great idea (more then just helium plasma )

**Comment by botn365** [comment]
fuel consumption is for all the fuels the same

**Comment by GeicowithaBango** [comment]
so you would need 5kL of fuel or helium plasma to fly to the moon?

**Review by draknyte1 — APPROVED** [review]

File: src/main/java/micdoodle8/mods/galacticraft/core/recipe/RecipeManagerGC.java (hunk 2 above)
Comment (bartimaeusnek) [reply src/main/java/micdoodle8/mods/galacticraft/core/recipe/RecipeManagerGC.java:68]: Cetane-Boosted Diesel, HOG, maybe a Plasma, EnderIO "Rocket Fuel", etc.

Comment (botn365) [reply src/main/java/micdoodle8/mods/galacticraft/core/recipe/RecipeManagerGC.java:68]: dont realy see the point of the eio rocket fuel sins the recipe is removed

**Comment by 0lafe** [comment]
What's wrong with t1-2 needing rp1, 3-4 using hydrazine, 5-6 purple fuel and 7-8 green?

**Comment by botn365** [comment]
for bart becous gt++ is arr

**Comment by 0lafe** [comment]
Sure but rocket fuels seem like a better option than anything else suggested, but also due to that would be almost useless to use due to their complexity. No reason to make green fuel, when you can easily grab a few b of plasma and throw that in

**Comment by draknyte1** [comment]
The fact it’s ARR shouldn’t diffuse the point of the pre-existing content. 🤦🏻‍♀️
What’s @Dream-Master think about easier fuels? Last I checked he wanted them as the tiered fuels.

**Comment by Regian24** [comment]
What about making fuel tiers? so it's like you need 100,000L of diesel to fly to the moon, 50,000L of cetane diesel, and like 10,000 of rocket fuel

**Comment by Dream-Master** [comment]
@Regian24  Diesel really? Show me a rocket with Diesel in rl :)
I would ban all low tier fuels because a rocket needs more power than a car.

**Comment by Regian24** [comment]
@Dream-Master It was just an example , i don't mean to actually have diesel in the rocket, it was just an example of the "Fuel tiering" system

**Comment by Regian24** [comment]
Maybe we can have Helium-3 as fuel in the rocket because it's abundant in the moon and it's actually used as fuel in rl

**Comment by botn365** [comment]
from my 5 second google search using helium-3 is teoreticly doable for rockets 
and it would be a realy boring fuel to use as well 
gt++ rocket fuels have a complex chain that needs to be made while helium 3 is just puting a oil drill on the moon

**Comment by Dream-Master** [comment]
@botn365 give us a config option so we can set fuels via cfg.

**Comment by GTNH-Colen** [comment]
> from my 5 second google search using helium-3 is teoreticly doable for rockets
> and it would be a realy boring fuel to use as well
> gt++ rocket fuels have a complex chain that needs to be made while helium 3 is just puting a oil drill on the moon

That's all well and good but it's essentially a nerf in disguise as people will have to set up huge complicated chains just so they can use the rocket they made.

**Comment by GeicowithaBango** [comment]
And we don’t  want that?

**Comment by 0lafe** [comment]
We definitely want that

**Comment by draknyte1** [comment]
@GTNH-Colen a nerf? It’s a feature dream wanted a loooong time ago.

**Comment by Dream-Master** [comment]
@draknyte1 yes it is true i want rocket fuels long time ago be the only fuels for rockets. Idk the recipes yet so need to check it

**Comment by GTNH-Colen** [comment]
> We definitely want that

"We" being your anti fun, anti GT++, anti QoL squad? 

> @GTNH-Colen a nerf? It’s a feature dream wanted a loooong time ago.

Maybe so but it's still a nerf if this gets implemented?

We need to open this up to a proper RFC thread in the main github before making any major changes like that.

**Comment by GeicowithaBango** [comment]
Im afraid I have to ask.
if you aren’t considering building and setting up bases as something pleasurable are you sure that you are playing the right game ?

**Comment by Prometheus0000** [comment]
Could you maybe specify what the fuel changes are exactly? Are they just more efficient, gating, are we using plasma or not, what are they (the actual names), and if they're gating what tier of rocket need what. I'm not opposed to using rocket fuels or gating rockets behind making them, but it's not clear from this discussion what the changes exactly are.

**Comment by Prometheus0000** [comment]
Could you maybe specify what the fuel changes are exactly? Are they just more efficient, gating, are we using plasma or not, what are they (the actual names), and if they're gating what tier of rocket need what. I'm not opposed to using rocket fuels or gating rockets behind making them, but it's not clear from this discussion what the changes exactly are.

**Comment by Prometheus0000** [comment]
So, no answer? How are people supposed to know what fuels they can use if you don't tell them in an interface or add the info to a quest? And I can't do that if you don't tell me, you know? Are you planning on doing so? Do you not expect people to get confused when their rockets don't work?

**Comment by botn365** [comment]
the current changes are 
diesel cant be used any  more and is replaced by the gt++ rocket fuels
rp-1 can be used up to T2 rocket
denshydrazine can be used up to T4 rocket
purple rocket fuel (CN3H7O3) can be used up to T6 rocket
green rocket fuel (H8N4C2O4) can be used in all rockets

**Comment by Prometheus0000** [comment]
When you say 'up to T2' that means _not_ including T2. Presumably you mean up to and _including_ T2?

I saw the EIO rocket fuel listed, but there was a comment that there's no recipe. Is there going to be a recipe added (and kick the GT++ ones up a tier maybe) or should that be removed? 

Do you use less of a higher tier fuel in a lower tier rocket?

What is a launch controller? Is it different from a launch pad? You teleport if you use it, but what does that mean exactly? You 'reduce' (not skip?) the countdown and the launching from the planet, but do you land instantly too? Is it available right away, or do you need to get some stuff from other planets to make it? And can/do you prevent people from _not_ using it? Is it obvious how to use?

**Comment by botn365** [comment]
including
eio is more of a backup incase of gt++ problems
at the moment there is no difrent fuel usage depnding on the fuel
the launch controller is somthing you place next to a launch pad in both locations
and yes it will instently land the rocket 
does also consume a lot of fuel 50% of the tank
needs desh so is T2

**Comment by Prometheus0000** [comment]
I don't play multiplayer, but is there a way to determine where it'll land in mp? Like, not at someone else's launch pad for example.

**Comment by botn365** [comment]
now that i think about it you can perfectly land at somone elses launch pad

**Comment by Prometheus0000** [comment]
Uh, I just checked the recipe for RP-1, and it says it's T3 (and the others are too), which means you need to make the frame of the Chemical Plant out of Titanium. Before you go to the Moon. Even a conservative estimate will reach into the hundreds of ingots required. Or will the recipe tier be different in beta3?

**Comment by botn365** [comment]
T3 is stainless but that is a lot of stainlesssteel to get

**Comment by draknyte1** [comment]
Least stainless isn't super hard to make.

**Comment by Prometheus0000** [comment]
Isn't it 1 = steel, 2= ssteel, 3 = titanium, 4 = tsteel?

**Comment by draknyte1** [comment]
You'd be correct actually.

Will adjust the fuel tiers a touch.

**Comment by botn365** [comment]
huh i tought aluminium was a tier

**Comment by Prometheus0000** [comment]
BTW are there still recipes you have yet to migrate to the CP? T4 wasn't listed as a requirement for any of them.

**Comment by draknyte1** [comment]
T3 is tungsten. 
T0 is bronze. 

Maybe should bump that to +1 in NEI for better understandability.

**Comment by Prometheus0000** [comment]
? Your own image doesn't list bronze casing? Just a bronze pipe. Did you change it? And there's no T0 recipes? They already start at 1 in NEI so bumping it up would make the recipes list 2-4?
https://cdn.discordapp.com/attachments/224191655375273985/663064109058621491/image0.jpg
This is the quest currently:
Untitled

So you would need to make the CP out of TSteel to make even the first rocket fuel currently then? You said in discord you moved RP-1 down a tier, but that would just make RP-1 go from TSteel to Titanium?

**Comment by draknyte1** [comment]
Hrmm, I’ll recheck. Don’t remember off the top of my head.

**Comment by draknyte1** [comment]
I'm going to redo the solid casings a little bit. By adding more tiers, both lower and higher.
WFe50 will become T6, while there will be another 2 tiers beyond that.

T1 = Bronze
T2 = Steel
T3 = Aluminium
T4 = Stainless
T5 = Titanium
T6 = TungstenSteel
T7/T8 = Undecided yet

This makes the Chem plant available a little bit earlier. (Albeit rather shit at T1)
Gives me more room to space out recipes as well.

**Pull request merged**